smart-py/marketplace.py

```diff
@@ -1,65 +1,954 @@
-        
+import smartpy as sp
+
 class Marketplace(sp.Contract):
     def __init__(self, objkt, metadata, manager, fee):
         self.init(
             objkt = objkt,
             metadata = metadata,
             manager = manager,
-            swaps = sp.big_map(tkey=sp.TNat, tvalue=sp.TRecord(issuer=sp.TAddress, objkt_amount=sp.TNat, objkt_id=sp.TNat, xtz_per_objkt=sp.TMutez, royalties=sp.TNat, creator=sp.TAddress)),
+            swaps = sp.big_map(
+                tkey=sp.TNat,
+                tvalue=sp.TRecord(
+                    issuer=sp.TAddress,
+                    objkt_amount=sp.TNat,
+                    objkt_id=sp.TNat,
+                    xtz_per_objkt=sp.TMutez,
+                    royalties=sp.TNat,
+                    creator=sp.TAddress
+                )
+            ),
             counter = 500000,
             fee = fee
-            )
-            
+        )
+
     @sp.entry_point
     def swap(self, params):
```

Comment:
Don't understand how the called knows the swap_id of the objkt just swapped? Is the assumption counter-1? Are there any potential concurrency issues to worry about? Would an explicit swap_id assignment be better?

Reply:
i'd personally like some sort of id issuing contract whose sole function is to issue unique autoincrementing objkt ids

regarding knowing the swap id that's something i'll have to look into

Reply:
additionally, i believe there have been isolated reports of race condition type bugs, i believe this can be resolved with a component that issues ids instead of the contract tracking its own

i looked into the swap_id, this is something that's inside the objkt_swap.py file - i don't know if these are 2 different contracts that work together? in the documentation there are several contract addresses provided, but i've (intentionally) never looked at the main site code